src/vs/workbench/browser/layout.ts

```diff
@@ -366,6 +392,49 @@ export abstract class Layout extends Disposable implements IWorkbenchLayoutServi
 		this.layout();
 	}
 
+	private setWindowBorder(skipLayout: boolean = false) {
+		if (isWeb || getTitleBarStyle(this.configurationService, this.environmentService) !== 'custom') {
+			return;
+		}
+
+		const theme = this.themeService.getTheme();
+
+		const activeBorder = theme.getColor(WINDOW_ACTIVE_BORDER);
+		const inactiveBorder = theme.getColor(WINDOW_INACTIVE_BORDER);
+
+		let windowBorder = false;
+		if (activeBorder || inactiveBorder) {
+			windowBorder = true;
+
+			let borderColor = this.state.hasFocus ? activeBorder : inactiveBorder;
+			if (!borderColor) {
+				borderColor = theme.getColor(this.state.hasFocus ? TITLE_BAR_ACTIVE_BACKGROUND : TITLE_BAR_INACTIVE_BACKGROUND);
+			}
+			this.container.style.setProperty('--window-border-color', borderColor ? borderColor.toString() : 'transparent');
```

Comment:
All in all I think I would really not do any border business if no color is defined.

Reply:
I removed the fallback to the titlebar color, and instead if one color is missing, it will just fallback to the other one.